santoku/salesforce/objects_handler.py

```diff
@@ -370,40 +398,48 @@ def do_request(self, method: str, path: str, payload: Optional[Dict[str, str]] =
 
         Raises
         ------
-        AssertionError
-            If the method is not supported, or the object is not valid, or payload is missing when
-            needed.
+        SalesforceObjectFieldError
+            If any field in the payload is invalid, any required field is empty or missing.
+
+        SalesforceObjectError
+            If the object in the query is not a valid salesforce object.
+
+        RequestMethodError
+            If the method is not supported, or the payload is missing when needed.
 
         requests.exceptions.RequestException
             If the connection with salesforce fails, e.g. the requesting resource does not exist.
 
         """
-        assert method in ["POST", "GET", "PATCH", "DELETE"], "Method isn't supported."
+        if method not in ["POST", "GET", "PATCH", "DELETE"]:
+            raise RequestMethodError("Method isn't supported.")
 
         if not self._is_authenticated:
             self._authenticate()
 
         if self._validate_salesforce_object:
-            salesforce_object_name = self._obtain_salesforce_object_name_from_path(path)
-            if salesforce_object_name:
-                assert (
-                    salesforce_object_name in self.get_salesforce_object_names()
-                ), f"{salesforce_object_name} isn't a valid object"
+            path_salesforce_object = self._obtain_salesforce_object_name_from_path(path)
+            if path_salesforce_object:
+                if path_salesforce_object.upper() not in (
+                    object_name.upper() for object_name in self.get_salesforce_object_names()
+                ):
+                    raise SalesforceObjectError(f"{path_salesforce_object} isn't a valid object")

```

Comment:
Would appreciate a comment somewhere appropriate explaining that Salesforce is case insensitive when using SOQL (thus comparing in all upper is fine).